### PR TITLE
feat(MainchainGateway): rollback deposit WETH, depositBatch

### PR DESCRIPTION
**Pull request by nxqbao**
### Description
- rollback impl of deposit WETH
- remove requestDepositForBatch

### CODE REVIEW

**Review by khanhtaskymaviscom — APPROVED** [review]

File: src/mainchain/MainchainGatewayV3.sol [hunk 1]
```diff
@@ -373,11 +363,16 @@ contract MainchainGatewayV3 is
...
+
+      /**
+       * Withdraw if token is WETH
+       *
+       * `IWETH.withdraw` only sends 2300 gas, which might be insufficient when recipient is a proxy, in this case, gateway proxy.
+       * However, the storage accesses on Shanghai hardfork are warm-access, only requires additional 100*2 gas. So it should be safe,
+       * no need to go via a mediator of WETH unwrapper.
+       */
       if (_roninWeth == _request.tokenAddr) {
-        wrappedNativeToken.approve(address(wethUnwrapper), _request.info.quantity);
```
Comment (khanhtaskymaviscom) [inline src/mainchain/MainchainGatewayV3.sol:379]: should deprecate `wethUnwrapper` variable as well

**Review by khanhtaskymaviscom — APPROVED** [review]

File: src/mainchain/MainchainGatewayV3.sol [hunk 2]
```diff
@@ -51,7 +51,8 @@ contract MainchainGatewayV3 is
 
   uint96 private _totalOperatorWeight;
   mapping(address operator => uint96 weight) private _operatorWeight;
-  WethUnwrapper public wethUnwrapper;
+  /// @custom:deprecated Previously `_wethUnwrapper` (uint256)
```
Comment (TuDo1403) [inline src/mainchain/MainchainGatewayV3.sol]:
```suggestion
  /// @custom:deprecated Previously `_wethUnwrapper` (address)
```

File: src/mainchain/MainchainGatewayV3.sol [hunk 3]
```diff
@@ -128,7 +129,10 @@ contract MainchainGatewayV3 is
   }
 
   function initializeV4(address payable wethUnwrapper_) external reinitializer(4) {
```
Comment (TuDo1403) [inline src/mainchain/MainchainGatewayV3.sol]:
```suggestion
  function initializeV4(address payable /* wethUnwrapper_ */) external reinitializer(4) {
```

File: src/mainchain/MainchainGatewayV3.sol [hunk 4]
```diff
@@ -373,11 +367,16 @@ contract MainchainGatewayV3 is
...
 
       _request.info.handleAssetIn(_requester, _request.tokenAddr);
-      // Withdraw if token is WETH
-      // The withdraw of WETH must go via `WethUnwrapper`, because `WETH.withdraw` only sends 2300 gas, which is insufficient when recipient is a proxy.
+
+      /**
+       * Withdraw if token is WETH
+       *
+       * `IWETH.withdraw` only sends 2300 gas, which might be insufficient when recipient is a proxy, in this case, gateway proxy.
+       * However, the storage accesses on Shanghai hardfork are warm-access, only requires additional 100*2 gas. So it should be safe,
```
Comment (TuDo1403) [inline src/mainchain/MainchainGatewayV3.sol]:
```suggestion
       * However, the storage accesses of proxy relating variables on Shanghai hardfork are warm-access, only requires additional 100*2 gas. So it should be safe,
```

**Review by khang-sm — APPROVED** [review]

File: src/mainchain/MainchainGatewayV3.sol [hunk 5]
```diff
@@ -373,11 +367,16 @@ contract MainchainGatewayV3 is
...
+       * no need to go via a mediator of WETH unwrapper.
+       */
       if (_roninWeth == _request.tokenAddr) {
-        wrappedNativeToken.approve(address(wethUnwrapper), _request.info.quantity);
-        wethUnwrapper.unwrap(_request.info.quantity);
+        IWETH(_roninWeth).withdraw(_request.info.quantity);
```
Comment (TuDo1403) [inline src/mainchain/MainchainGatewayV3.sol]:
```suggestion
        IWETH(wrappedNativeToken).withdraw(_request.info.quantity);
```

**Review by khanhtaskymaviscom — APPROVED** [review]

**Review by TuDo1403 — APPROVED** [review]
LGTM.

**Pull request merged**